src/components/CreateCriteriaForm/CreateCreteriaForm.tsx

```diff
@@ -0,0 +1,272 @@
+            </ContentItem>
+            <TitleItem>
+                <TitleContainer>
+                    <Title>{props.title}</Title>
+                </TitleContainer>
+            </TitleItem>
+            <ContentItem>
+                {nullable(props.state, (s) => (
+                    <StateDot size="m" title={s?.title} hue={s?.hue} />
+                ))}
+            </ContentItem>
+            <ContentItem>
+                <TextItem>{props.projectId}</TextItem>
+            </ContentItem>
+        </TableRow>
+    );
+};
+
+interface WeightFieldProps {
+    registerProps: UseFormRegisterReturn<'weight'>;
+    errorsResolver: (field: 'weight') => { message?: string } | undefined;
+    maxValue: number;
+    setError: UseFormSetError<AddCriteriaScheme>;
+}
+
+const WeightField: React.FC<WeightFieldProps> = ({ registerProps, errorsResolver, maxValue, setError }) => {
+    const handleChange = useCallback<React.ChangeEventHandler<HTMLInputElement>>(
+        (event) => {
+            const { value } = event.target;
+
+            const parsedValue = parseInt(value, 10);
+            let message: string | undefined;
+
+            if (!value) {
+                message = tr('Weight is required');
+            } else if (Number.isNaN(parsedValue)) {
+                message = tr('Weight must be integer');
+            } else if (parsedValue <= 0 || maxValue + parsedValue > maxPossibleWeigth) {
+                message = tr
+                    .raw('Weight must be in range', {
+                        upTo: `${maxPossibleWeigth - maxValue}`,
+                    })
+                    .join('');
+            }
+
+            setError('weight', { message });
+        },
+        [setError, maxValue],
+    );
+
+    return (
+        <StyledFormInput
+            {...registerProps}
+            error={errorsResolver('weight')}
+            name="weight"
+            placeholder={tr('Weight')}
+            brick="center"
+            onChange={handleChange}
+        />
+    );
+};
+
+export const CreateCriteriaForm: React.FC<any> = ({ onSubmit, onSearch, items, goalId, sumOfWeights = 0 }) => {
+    const [[text, type], setQuery] = useState<[string, 'plain' | 'search']>(['', 'plain']);
+    const [selectedGoal, setSelectedGoal] = useState<(typeof items)[number]>(null);
+
+    const {
+        handleSubmit,
+        register,
+        reset,
+        setValue,
+        setError,
+        formState: { errors, isSubmitSuccessful },
+    } = useForm<AddCriteriaScheme>({
+        resolver: zodResolver(criteriaSchema),
+        mode: 'all',
+        reValidateMode: 'onChange',
+        criteriaMode: 'all',
+        defaultValues: {
+            linkedGoalId: goalId,
+            title: '',
+            weight: '',
+        },
+    });
+
+    const submitHandler: SubmitHandler<AddCriteriaScheme> = (values) => {
+        const data = { ...values };
+
+        if (selectedGoal) {
+            data.goalAsGriteria = selectedGoal.id;
```

Comment:
Not critical but I think we need to use hidden inputs for this cases everywhere.